### PR TITLE
MAO: add e2e-vsphere-serial manual run

### PR DESCRIPTION
**Pull request by michaelgugino**
Enable MAO repo to manually run e2e-vsphere-serial
job to assist in diagnosing and fixing bugs on
this platform.

### CODE REVIEW

**Review by elmiko — COMMENTED** [review]
this mostly makes sense to me, i just have a question about the workflow field in the config.

File: ci-operator/config/openshift/machine-api-operator/openshift-machine-api-operator-master.yaml [hunk 1]
```diff
@@ -110,6 +110,10 @@ tests:
   steps:
     cluster_profile: packet
     workflow: baremetalds-e2e
+- as: e2e-vsphere-serial
+  steps:
+    cluster_profile: vsphere
+    workflow: openshift-e2e-vsphere-serial
```
Comment (elmiko) [inline ci-operator/config/openshift/machine-api-operator/openshift-machine-api-operator-master.yaml:116]: do we need a workflow with this name somewhere?

Comment (elmiko) [reply ci-operator/config/openshift/machine-api-operator/openshift-machine-api-operator-master.yaml:116]: nvm, i found it!

**Comment by elmiko** [comment]
the test failures look like they are coming the actual test, not this change.
/retest

File: ci-operator/config/openshift/machine-api-operator/openshift-machine-api-operator-master.yaml (hunk 1 above)
Comment (michaelgugino) [reply ci-operator/config/openshift/machine-api-operator/openshift-machine-api-operator-master.yaml:116]: There is one, I think?  I don't know how this stuff is wired together.

Comment (elmiko) [reply ci-operator/config/openshift/machine-api-operator/openshift-machine-api-operator-master.yaml:116]: yeah there is, i found it here https://github.com/openshift/release/blob/master/ci-operator/step-registry/openshift/e2e/vsphere/serial/openshift-e2e-vsphere-serial-workflow.yaml

just took me a few to figure out where it was lol

**Pull request merged**